tests/unit/plugins/module_utils/test_attachment.py

```diff
@@ -0,0 +1,872 @@
+                    "type": "text/markdown",
+                    "hash": "hash",
+                }
+            )
+            == "attachment_name.txt"
+        )
+
+    def test_name_omitted(self):
+        assert (
+            attachment.get_file_name(
+                {
+                    "path": "some/path/file_name.txt",
+                    "name": None,
+                    "type": "text/markdown",
+                    "hash": "hash",
+                }
+            )
+            == "file_name.txt"
+        )
+
+
+class TestAttachmentGetFileType:
+    def test_type_specified(self):
+        assert (
+            attachment.get_file_type(
+                {
+                    "path": "some/path/file_name.txt",
+                    "name": "attachment_name.txt",
+                    "type": "text/markdown",
+                    "hash": "hash",
+                }
+            )
+            == "text/markdown"
+        )
+
+    def test_type_omitted(self):
+        assert (
+            attachment.get_file_type(
+                {
+                    "path": "some/path/file_name.txt",
+                    "name": "attachment_name.txt",
+                    "type": None,
+                    "hash": "hash",
+                }
+            )
+            == "text/plain"
+        )
+
+
+class TestAttachmentBuildQuery:
+    def test_name_type_specified(self):
+        assert attachment.build_query(
+            "table",
+            "1234",
+            {
+                "path": "some/path/file_name.txt",
+                "name": "attachment_name.txt",
+                "type": "text/markdown",
+                "hash": "hash",
+            },
+        ) == {
+            "file_name": "attachment_name.txt",
+            "content_type": "text/markdown",
+            "table_name": "table",
+            "table_sys_id": "1234",
+        }
+
+    def test_name_omitted(self):
+        assert attachment.build_query(
+            "table",
+            "1234",
+            {
+                "path": "some/path/file_name.txt",
+                "name": None,
+                "type": "text/markdown",
+                "hash": "hash",
+            },
+        ) == {
+            "file_name": "file_name.txt",
+            "content_type": "text/markdown",
+            "table_name": "table",
+            "table_sys_id": "1234",
+        }
+
+    def test_type_omitted(self):
+        assert attachment.build_query(
+            "table",
+            "1234",
+            {
+                "path": "some/path/file_name.txt",
+                "name": "attachment_name.txt",
+                "type": None,
+                "hash": "hash",
+            },
+        ) == {
+            "file_name": "attachment_name.txt",
+            "content_type": "text/plain",
+            "table_name": "table",
+            "table_sys_id": "1234",
+        }
+
+
+class TestAttachmentTransformMetadataList:
+    def test_normal(self, create_module):
```

Comment:
Pytest offers the `tmp_path` fixture that can make this a bit nicer compared to messing around with raw Python methods. See the remaining comments in this function for usage examples.

```suggestion
    def test_normal(self, create_module, tmp_path):
```